.ci/lint-cpp.sh

```diff
@@ -10,14 +10,11 @@ cpplint \
 echo "done running cpplint"
 
 echo "running cmakelint"
-cmake_files=$(
-    find . -name CMakeLists.txt -o -path "./cmake/*.cmake" \
-    | grep -v external_libs
-)
+cmake_files=$(find . \( -name CMakeLists.txt -o -path "./cmake/*.cmake" \) -not -path "./external_libs/*" | tr '\n' ' ')
```

Comment:
What was the purpose of these changes that you pushed after I approved? It looks to me like they result in the files we care about all being ignored.

```text
Ignoring file: ./cmake/IntegratedOpenCL.cmake ./cmake/Sanitizer.cmake ./cmake/modules/FindLibR.cmake ./CMakeLists.txt 
```

([build link](https://github.com/microsoft/LightGBM/actions/runs/10638315268/job/29504133350?pr=6621))

If this was from that `shellcheck` warning about assigning the result of `find` to a variable... let's just use `find -exec` instead. Please try this:

```shell
find \
    . \
    -type f \
    \( -name CMakeLists.txt -o -name '*.cmake' \) \
    -not -path './external_libs/*' \
    -exec cmakelint \
        --linelength=120 \
        --filter=-convention/filename,-package/stdargs,-readability/wonkycase \
        {} \+
```

I tested this locally (on my mac) and found it worked well. This listed the expected files:

```shell
find \
    . \
    -type f \
    \( -name CMakeLists.txt -o -name '*.cmake' \) \
    -not -path './external_libs/*'
```

```text
./CMakeLists.txt
./cmake/Sanitizer.cmake
./cmake/IntegratedOpenCL.cmake
./cmake/modules/FindLibR.cmake
```

And the full statement did raise warnings about CMake formatting when I changed some code in those scripts.

Reply:
@jameslamb First I did not mean to creep in a change after your approval, I am used to MR/PR settings in which a code change requires a re-approval. I changed the line because the `cmakelint` execution was failing the CI. I traced it to the file path arguments being passed with '\n' joins instead of single-whitespace. I changed the `find + (negative) grep` command for a more efficient `find` with a ignore path logic.

I missed the log line saying that the files were ignored, and I don't understand why it happens. I did think of uxing `exec` but I guessed the command would get quite long, hence less readable, and some one may want to print the value of `$cmake_files` for debugging purposes. I will accept your change, thanks for proofing!

Reply:
ohhhhhhh thank you so much for the thorough explanation!

There shouldn't be a `./build/` directory in CI (because we don't actually run a CMake build in the `lint` CI task), but it is likely that there could be one lying around when you run this locally if you've also been doing other development in the project.

Keeping `./cmake/*.cmake` for now is fine... this is how the linting here worked before (e.g. is not a change made by your PR) and all of the `*.cmake` files we currently want to lint do happen to be in that directory).